src/mon/Monitor.cc

```diff
@@ -3329,40 +3329,63 @@ void Monitor::handle_route(MonOpRequestRef op)
   }
 }
 
+namespace {
+  struct ReentrantLock {
+    const bool initially_locked;
+    Mutex &mutex;
+    ReentrantLock(Mutex& m)
+      : initially_locked(m.is_locked_by_me()),
```

Comment:
Better to make the caller(s) who don't already hold lock take it before calling resend_routed_requests?